File: pkg/rockferry/spec/cluster.go
package spec

type ClusterNodeKind string
type ClusterStatusState string

const (
	ClusterNodeKindWorker       ClusterNodeKind = "worker"
	ClusterNodeKindControlPlane                 = "control_plane"

	ClusterStatusStateCreating  ClusterStatusState = "creating"
	ClusterStatusStateHealthy                      = "healthy"
	ClusterStatusStateUpgrading                    = "upgrading"
)

type ClusterNodeSpec struct {
	Kind      ClusterNodeKind `json:"kind"`
	MachineId string          `json:"machine_id"`
}

type ClusterSpec struct {
	Name string `json:"name"`

	ControlPlaneConfig []byte `json:"control_plane_config"`
	WorkerConfig       []byte `json:"worker_config"`
	TalosConfig        []byte `json:"talos_config"`

	KubernetesVersion string `json:"kubernetes_version"`

	Nodes []*ClusterNodeSpec `json:"nodes"`
}

type ClusterStatus struct {
	State ClusterStatusState `json:"state"`
}

File: ui/src/types/cluster.ts
export enum ClusterNodeKind {
    Worker = "worker",
    ControlPlane = "control_plane",
}

export interface ClusterNode {
    machine_id: string;
    kind: ClusterNodeKind;
}

export interface Cluster {
    name: string;
    nodes: ClusterNode[];
    kubernetes_version: string;
    talos_config: string;
    control_plane_config: string;
    worker_config: string;
}

export enum ClusterStatusState {
    Creating = "creating",
    Healthy = "healthy",
}

export interface ClusterStatus {
    state: ClusterStatusState;
}
